conformance/lang/expressions/type_cast_expression.balt

```diff
@@ -0,0 +1,1698 @@
+Labels: array-type, byte, decimal, DecimalFloatingPointNumber, float, FloatingPointTypeSuffix, map-type, 
+        module-type-defn, record-type, type-cast-expr, union-type, xml
+
+type MyRecord record {|
+    int id;
+|};
+
+function init() {
+    float|string|MyRecord a = 3.44343;
+    decimal b = <decimal>a;
+    io:println(b); // @output 3.443430000000000212878603633726016
+
+    float|byte|MyRecord c = 5.5f;
+    int d = <int>c;
+    io:println(d); // @output 6
+    byte d0 = <byte>c;
+    io:println(d0); // @output 6
+
+    float|xml|int[] e = 324.7E-2f;
+    int f = <int>e;
+    io:println(f); // @output 3
+    byte f0 = <byte>e;
+    io:println(f0); // @output 3
+
+    decimal|map<int> g = 4444.4444;
+    io:println(g); // @output 4444.4444
+    float h = <float>g;
+    io:println(h); // @output 4444.4444
+
+    decimal|MyRecord|boolean i = 5.5d;
+    int j = <int>i;
+    io:println(j); // @output 6
+    byte j0 = <byte>i;
+    io:println(j0); // @output 6
+
+    decimal|float|MyRecord k = 324.7E-2d;
+    int m = <int>k;
+    io:println(m); // @output 3
+    byte m0 = <byte>k;
+    io:println(m0); // @output 3
+
+    int|float|string n = 54;
+    float p = <float>n;
+    io:println(p); // @output 54.0
+
+    int|MyRecord|string q = 949545456;
+    decimal r = <decimal>q;
+    io:println(r); // @output 949545456.0
+}
+
+Test-Case: output
+Description: Test numeric conversion of values by typecast expression from unions to unions.
+Labels: array-type, byte, decimal, DecimalFloatingPointNumber, float, FloatingPointTypeSuffix, map-type, 
+        module-type-defn, record-type, type-cast-expr, union-type, xml
+
+type MyRecord record {|
+    int id;
+|};
+
+function init() {
+    float|string|MyRecord a = 3.44343;
+    decimal|string b = <decimal|string>a;
+    io:println(b); // @output 3.443430000000000212878603633726016
+
+    float|byte|MyRecord c = 5.5f;
+    int|MyRecord d = <int|MyRecord>c;
+    io:println(d); // @output 6
+    byte|xml d0 = <byte|xml>c;
+    io:println(d0); // @output 6
+
+    float|xml|int[] e = 324.7E-2f;
+    int|int[] f = <int|int[]>e;
+    io:println(f); // @output 3
+    byte|json f0 = <byte|json>e;
+    io:println(f0); // @output 3.247
+
+    decimal|map<int> g = 4444.4444;
+    io:println(g); // @output 4444.4444
+    float|string h = <float|string>g;
+    io:println(h); // @output 4444.4444
+
+    decimal|MyRecord|boolean i = 5.5d;
+    int|map<int> j = <int|map<int>>i;
+    io:println(j); // @output 6
+    byte|boolean j0 = <byte|boolean>i;
+    io:println(j0); // @output 6
+
+    decimal|float|MyRecord k = 324.7E-2d;
+    int|string m = <int|string>k;
+    io:println(m); // @output 3
+    byte|boolean m0 = <byte|boolean>k;
+    io:println(m0); // @output 3
+
+    int|float|string n = 54;
+    float|string p = <float|string>n;
+    io:println(p); // @output 54.0
+
+    int|MyRecord|string q = 949545456;
+    decimal|string r = <decimal|string>q;
+    io:println(r); // @output 949545456.0
+}
```

Comment:
Since `byte` is equivalent to `int:Unsigned8` following should work. We can add a test for this
Created an issue for this https://github.com/ballerina-platform/ballerina-lang/issues/36418 
```ballerina
float x = 1;
int y = <byte|int:Unsigned8> x; // incompatible types: 'float' cannot be cast to '(byte|int:Unsigned8)'
```